WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift

```diff
@@ -30,6 +30,9 @@ final class POSEligibilityChecker: POSEligibilityCheckerProtocol {
                 .eraseToAnyPublisher()
         }
         return Publishers.CombineLatest(isWooCommerceVersionSupported, isPointOfSaleFeatureFlagEnabled)
+            .filter { [weak self] _ in
+                self?.isEligibleFromSiteChecks ?? false
```

Comment:
❓ Curious

I wonder if `filter` is needed here or not 🤔 I don't remember now what value it provides in `isOnboardingComplete`, do you remember?

Reply:
When the onboarding feature flag is disabled where we still require onboarding completed state for POS, the site checks are performed as a filter as well but on the onboarding state publisher:

https://github.com/woocommerce/woocommerce-ios/blob/e3755bd1a31b26ad19c2bb557895ee5041b077d5/WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift#L60-L70

I didn't want to change the behavior there to keep the scope small.

Lemme know if I might have misunderstood your question!